src/Core/src/Layouts/GridLayoutManager.cs

```diff
@@ -655,6 +662,70 @@ void ResolveStars(Definition[] defs, double availableSpace, Func<Cell, bool> cel
 				}
 			}
 
+			bool TryResolveStarsWithDensityAwareness(Definition[] defs, double availableSpace, double starCount)
+			{
+				// Try to get density from the Grid's MauiContext or handler
+				var density = GetDisplayDensity();
+				if (density <= 0)
+				{
+					return false; // No density available, fall back to regular method
+				}
+
+				// Create portions array for star definitions
+				var starDefs = new List<Definition>();
+				var portions = new List<double>();
+
+				foreach (var def in defs)
+				{
+					if (def.IsStar)
+					{
+						starDefs.Add(def);
+						portions.Add(def.GridLength.Value);
+					}
+				}
+
+				if (starDefs.Count == 0)
+					return true; // No star definitions to resolve
+
+				// Use DensityValue to distribute pixels precisely
+				var pixelAllocations = DensityValue.DistributePixels(availableSpace * density, density, portions.ToArray());
+
+				// Convert back to dp and assign to definitions
+				for (int i = 0; i < starDefs.Count; i++)
+				{
+					starDefs[i].Size = pixelAllocations[i] / density;
+				}
+
+				return true;
+			}
+
+			double GetDisplayDensity()
```

Comment:
Create an extention method off of IWindow that's able to get the DisplayDensity



You'll have to create a separate implementation for each platform with how this is retrieved